numpyro/handlers.py

```diff
@@ -596,7 +599,7 @@ def __init__(self, fn=None, data=None, substitute_fn=None):
         super(substitute, self).__init__(fn)
 
     def process_message(self, msg):
-        if (msg['type'] not in ('sample', 'param')) or msg.get('_control_flow_done', False):
+        if (msg['type'] not in ('sample', 'param', 'plate')) or msg.get('_control_flow_done', False):
```

Comment:
I added `plate` here so we can substitute if needed. This is useful for #724 I guess. The behavior should be similar to the way we use `plate(..., subsample=)` in Pyro.